src/client.ts

```diff
@@ -88,28 +91,66 @@ export class SkynetClient {
   openFileHns = openFileHns;
   resolveHns = resolveHns;
 
+  // MySky
+
+  extractDomain = extractDomain;
+  getFullDomainUrl = getFullDomainUrl;
+  loadMySky = loadMySky;
+
   // SkyDB
+
   db = {
     getJSON: getJSON.bind(this),
     setJSON: setJSON.bind(this),
+
+    // Helpers.
+    getOrCreateRegistryEntry: getOrCreateRegistryEntry.bind(this),
   };
 
   // SkyDB helpers
+
   registry = {
     getEntry: getEntry.bind(this),
     getEntryUrl: getEntryUrl.bind(this),
     setEntry: setEntry.bind(this),
+
+    // Helpers.
+    postSignedEntry: postSignedEntry.bind(this),
+    signEntry: signEntry.bind(this),
   };
 
   /**
    * The Skynet Client which can be used to access Skynet.
    *
    * @class
-   * @param [portalUrl] The portal URL to use to access Skynet, if specified. To use the default portal while passing custom options, use ""
+   * @param [initialPortalUrl] The initial portal URL to use to access Skynet, if specified. A request will be made to this URL to get the actual portal URL. To use the default portal while passing custom options, pass "".
    * @param [customOptions] Configuration for the client.
    */
-  constructor(portalUrl: string = defaultPortalUrl(), customOptions: CustomClientOptions = {}) {
-    this.portalUrl = portalUrl;
+  constructor(initialPortalUrl = "", customOptions: CustomClientOptions = {}) {
+    // Get the initial portal URL, if not provided.
+    if (initialPortalUrl == "") {
+      initialPortalUrl = defaultPortalUrl();
+    }
+    // Kick off async request for the portal URL.
+    this.portalUrl = this.executeRequest({
+      ...customOptions,
+      method: "head",
+      url: initialPortalUrl,
+      endpointPath: "/",
+    }).then((response) => {
+      /* istanbul ignore next */
+      if (typeof response.headers === "undefined") {
+        throw new Error(
```

Comment:
throwing here (2x) results in `error in uncaught promise` I believe

Reply:
Already addressed.